packages/clay-color-picker/src/index.tsx

```diff
@@ -280,44 +290,43 @@ const ClayColorPicker: React.FunctionComponent<IProps> = ({
 									value,
 								])}
 								disabled={disabled}
-								insetBefore
+								insetBefore={!!isHex}
 								onBlur={(event) => {
-									const newColor = tinycolor(
-										event.target.value
-									);
+									if (!isColorValid(event.target.value)) {
+										setInvalidColor(true);
+									}
 
-									const hexString = newColor.isValid()
-										? newColor.toHex()
-										: event.target.value;
+									onValueChange(event.target.value);
 
-									onValueChange(hexString);
-									setHexInputValue(hexString);
 									onBlur(event);
 								}}
 								onChange={(event) => {
-									const newHexValue = event.target.value;
-
-									const newColor = tinycolor(newHexValue);
-
-									setHexInputValue(newHexValue);
-
-									if (newColor.isValid()) {
-										onValueChange(newColor.toHex());
+									if (!isColorValid(event.target.value)) {
+										setInvalidColor(false);
 									}
+
+									onValueChange(event.target.value);
 								}}
 								ref={inputRef}
 								type="text"
-								value={hexInputValue
-									.toUpperCase()
-									.substring(0, 6)}
+								value={value}
 							/>
-
-							<ClayInput.GroupInsetItem before tag="label">
-								{'#'}
-							</ClayInput.GroupInsetItem>
+							{isHex && (
+								<ClayInput.GroupInsetItem before tag="label">
+									{'#'}
+								</ClayInput.GroupInsetItem>
+							)}
 						</ClayInput.GroupItem>
 					)}
 				</ClayInput.Group>
+
+				{invalidColor && (
+					<ClayForm.FeedbackGroup>
+						<ClayForm.FeedbackItem>
+							{feedbackText}
+						</ClayForm.FeedbackItem>
+					</ClayForm.FeedbackGroup>
+				)}
```

Comment:
@bryceosterhaus I didn't get to check it out but whoever uses the ColorPicker will probably have to know when it is invalid to correctly format the validation markup as shown here https://next.clayui.com/docs/components/form.html#validation.

I think it would be better to expose an `onValidation` to inform which error happened and let the developer take care of that mainly so we don't start adding more use cases within that component. What do you think?

Reply:
@matuzalemsteles oh yeah thats a great idea, i'll switch that up to be a callback and allow them to insert their own content.

Reply:
Actually, since the user has the `onChange` callback, there is no need to provide internal validation. The user can decide what is a valid color, rather than us deciding that for them